File: tutorials/testing/python/array1d.py
from bridges.bridges import *
from bridges.array import *
from bridges.array1d import *
import sys


def main():

    # create the Bridges object, set credentials
    bridges = Bridges(ASSIGNMENT_NUMBER, "USER_ID", "API_KEY")


	# Bridges title, description
    bridges.set_title("One-Dimensional Array Example")
    bridges.set_description("Illustrates a one dimensional array with visual attributes")

    #set array dimensions, allocate array of elements
    arraySize = 10
    arr = Array1D(arraySize)

    #populate the array with squares of indicies
    for i in range(arr.size):
        arr[i].label = i*i

    #color the array elements
    arr[0].visualizer.color = "red"
    arr[1].visualizer.color = "green"
    arr[2].visualizer.color = "blue"
    arr[3].visualizer.color = "cyan"
    arr[4].visualizer.color = "magenta"
    arr[5].visualizer.color = "yellow"
    arr[6].visualizer.color = "red"
    arr[7].visualizer.color = "green"
    arr[8].visualizer.color = "blue"
    arr[9].visualizer.color = "black"

    # tell Bridges what data structure to visualize
    bridges.set_data_structure(arr)

    # visualize the list
    bridges.visualize()

if __name__ == "__main__":
    main()

File: tutorials/testing/c++/array1d.cpp
#include "Bridges.h"
#include "Array1D.h"

using namespace bridges;

int main(int argc, char **argv) {
	// create Bridges object
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", "YOUR_API_KEY");
	// title, description
	bridges.setTitle("One-Dimensional Array Example");
	bridges.setDescription("Illustrates a one dimensional array with visual attributes");

	// create, populate the array
	Array1D<int> arr (10);

	// populate the array, with squares of indices
	// use the values to label the elements
	for (int j = 0; j < 10; j++) {
		arr[j] = j * j;
		arr.getElement(j).setLabel(to_string(arr[j]));
	}

	// set visual attributes
	arr.getElement(0).setColor("red");
	arr.getElement(1).setColor("green");
	arr.getElement(2).setColor("blue");
	arr.getElement(3).setColor("cyan");
	arr.getElement(4).setColor("magenta");
	arr.getElement(5).setColor("yellow");
	arr.getElement(6).setColor("red");
	arr.getElement(7).setColor("green");
	arr.getElement(8).setColor("blue");
	arr.getElement(9).setColor("black");

	// visualize
	bridges.setDataStructure(&arr);
	bridges.visualize();

	return 0;
}
